From 761ce3700be488f3967ac024fbd33b09f54a2962 Mon Sep 17 00:00:00 2001
From: Marek Marczykowski <marmarek@invisiblethingslab.com>
Date: Sun, 15 Jul 2012 19:57:47 +0200
Subject: [PATCH] pvops/xen-blkfront: handle FDEJECT as detach request (#630)

---
 drivers/block/xen-blkfront.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/block/xen-blkfront.c b/drivers/block/xen-blkfront.c
index a9ba7ec4d1e5..034716c61e43 100644
--- a/drivers/block/xen-blkfront.c
+++ b/drivers/block/xen-blkfront.c
@@ -48,6 +48,7 @@
 #include <linux/list.h>
 #include <linux/workqueue.h>
 #include <linux/sched/mm.h>
+#include <linux/fd.h>
 
 #include <xen/xen.h>
 #include <xen/xenbus.h>
@@ -512,6 +513,9 @@ static int blkif_ioctl(struct block_device *bdev, fmode_t mode,
 			return 0;
 		return -EINVAL;
 	}
+	case FDEJECT:
+		xenbus_switch_state(info->xbdev, XenbusStateClosing);
+		return 0;
 
 	default:
 		/*printk(KERN_ALERT "ioctl %08x not supported by Xen blkdev\n",
-- 
2.25.3

