--- linux-2.6.38.3/drivers/xen/blkback/cdrom.c.orig	2011-07-24 18:44:14.795354823 +0200
+++ linux-2.6.38.3/drivers/xen/blkback/cdrom.c	2011-07-24 18:45:46.278542130 +0200
@@ -35,9 +35,15 @@
 #include "common.h"
 
 #undef DPRINTK
+
+#undef DEBUG
+#ifdef DEBUG
 #define DPRINTK(_f, _a...) \
 	printk(KERN_DEBUG "(%s() file=%s, line=%d) " _f "\n", \
 	       __func__, __FILE__ , __LINE__ , ##_a )
+#else
+#define DPRINTK(_f, _a...)
+#endif
 
 
 #define MEDIA_PRESENT "media-present"
